doc/BGP/Bgpcfgd-dyn-peer-modification-support.md

```diff
@@ -0,0 +1,145 @@
+ 
+  * [1 Requirements Overview](#1-requirements-overview)
+    * [1.1 Functional requirements](#11-functional-requirements)
+    * [1.2 Scalability requirements](#12-scalability-requirements)
+  * [2 Modules Design](#2-modules-design)
+    * [2.1 Config DB](#21-config-db)
+    * [2.2 State DB](#22-state-db)
+    * [2.3 CLI](#23-cli)
+    * [2.4 Bgpcfgd](#24-bgpcfgd)
+    * [2.5 Docker-FPM-FRR bgpd templates](#25-docker-fpm-frr-bgpd-template)
+  * [3 Test Plan](#3-test-plan)
+
+###### Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 04/09/2025  |    Anish Narsian   | Initial version                   |
+| 1.0 | 07/02/2025  |    Anish Narsian   | All comments addressed            |
+
+# About this Manual
+This document provides the high level design to support enhanced CRUD operations on dynamic bgp peers(ie: bgp listen ranges) using a standard template-based mechanism, as well as the CLI design for some useful per VRF BGP commands.
+
+# Use case
+SDN(Software Defined Networking) is used to program SONiC devices for various use cases, and we intend to extend this to BGP configurations, specifically dynamic BGP peers. To that extent we intend to support enhanced CRUD operations on dynamic peer types, including modifying peer ranges and supporting modification and deletion of their associated configurations like route maps, prefix lists and peer groups using the standard template mechanism.
+
+
+# Definitions/Abbreviation
+###### Table 1: Abbreviations
+| Abbreviation             | Meaning                        |
+|--------------------------|--------------------------------|
+| BGP                      | Border Gateway Protocol        |
+| BGPCFGD                  | BGP configuration daemon       |
+| VNET                     | Virtual Network                |
+| VRF                      | Virtual Routing and Forwarding |
+
+
+# 1 Requirements Overview
+## 1.1 Functional requirements
+This section describes the SONiC requirements for supporting enhanced CRUD operations for dynamic bgp peers:
+- Configurable in VNET/VRFs, but equally applicable to default VRF
+- Dynamic peers' listen ranges must be modifiable at runtime such that ranges can be added and removed(changing today's behavior of being a create only attribute)
+- Dynamic peer updates/deletes should allow for changes to associated configurations like route-maps, peer-groups, prefix-lists etc
+- Ability to track the configuration of a dynamic BGP peer via a State DB entry which is queryable by the SDN controller
+- CLI commands to show VRF/VNET BGP neighbors
+
+## 1.2 Scalability requirements
+###### Table 2: Scalability
+| Component                | Expected value              |
+|--------------------------|-----------------------------|
+| Number of Dynamic BGP Peers each with route-map, prefix-list and peer-group|2k|
```

Comment:
@anish-n Could you clarify what the "2K" refers to in this context? A scale of 2,000 is significantly larger compared to 64 neighbors and could potentially lead to performance issues or even system crashes. I recommend explicitly defining what the 2K and 4K entries represent in the table to avoid confusion.